tcs-service/src/test/java/com/transformuk/hee/tis/tcs/service/api/CurriculumMembershipResourceTest.java

```diff
@@ -129,6 +131,49 @@ void shouldCreateCurriculumMembership() throws Exception {
         .andExpect(jsonPath("$.curriculumId").value(curriculumId));
   }
 
+  @Test
+  @Transactional
+  void shouldPatchCurriculumMembership() throws Exception {
+    personRepository.saveAndFlush(person);
+    curriculum.setStatus(Status.CURRENT);
+    cmRepository.saveAndFlush(curriculum);
+    programme.setCurricula(Collections.singleton(programmeCurriculum));
+    programmeRepository.saveAndFlush(programme);
+    programmeMembership.setProgramme(programme);
+    programmeMembership.setProgrammeStartDate(START_DATE_1);
+    programmeMembership.setProgrammeEndDate(END_DATE_1);
+
+    CurriculumMembership curriculumMembership = new CurriculumMembership();
+    curriculumMembership.setCurriculumId(curriculum.getId());
+    curriculumMembership.setProgrammeMembership(programmeMembership);
+    curriculumMembership.setCurriculumStartDate(START_DATE_1);
+    curriculumMembership.setCurriculumEndDate(END_DATE_1);
+    programmeMembership.setCurriculumMemberships(Collections.singleton(curriculumMembership));
+
+    programmeMembershipRepository.saveAndFlush(programmeMembership);
+    curriculumMembershipRepository.saveAndFlush(curriculumMembership);
+
+    CurriculumMembershipDTO cmDto = new CurriculumMembershipDTO();
+    Long curriculumId = programme.getCurricula().iterator().next().getCurriculum().getId();
+    cmDto.setId(curriculumMembership.getId());
+    cmDto.setCurriculumId(curriculumId);
+    cmDto.setCurriculumStartDate(START_DATE_2);
+    cmDto.setCurriculumEndDate(END_DATE_2);
+    cmDto.setProgrammeMembershipUuid(programmeMembership.getUuid());
+
+    restCmMockMvc.perform(patch("/api/curriculum-memberships")
+            .contentType(MediaType.APPLICATION_JSON)
+            .content(TestUtil.convertObjectToJsonBytes(cmDto)))
+        .andDo(print())
+        .andExpect(status().isOk())
+        .andExpect(jsonPath("$.id").isNotEmpty())
+        .andExpect(jsonPath("$.curriculumStartDate").value(START_DATE_2.toString()))
+        .andExpect(jsonPath("$.curriculumEndDate").value(END_DATE_2.toString()))
+        .andExpect(
+            jsonPath("$.programmeMembershipUuid").value(programmeMembership.getUuid().toString()))
+        .andExpect(jsonPath("$.curriculumId").value(curriculumId));
+  }
+
```

Comment:
I think it would be better if there's one more integration test for the case when validation fails.